neqo-transport/src/connection/mod.rs

```diff
@@ -2491,6 +2530,18 @@ impl Connection {
                 self.acks
                     .ack_freq(seqno, tolerance - 1, delay, ignore_order);
             }
+            Frame::Datagram { data, .. } => {
+                let max_size = self
+                    .tps
+                    .borrow()
+                    .local
+                    .get_integer(tparams::MAX_DATAGRAM_FRAME_SIZE);
+                if max_size < u64::try_from(data.len()).unwrap() {
+                    return Err(Error::ProtocolViolation);
+                }
+                self.events.datagram(data);
```

Comment:
The copy that happens here is a little unfortunate.  It might be a good idea to mark this for future investigation.  I don't think that a copy is going to be so bad for us, but we might want to enable a more efficient handling process.  That's a project though, so this is fine.

Reply:
I have already spent some time thinking about this. I do not have a solution at the moment. Especially with rust-c++ glue code this will need another copy. I will open an issue, maybe we will have a better idea.